tailwind.config.js

```diff
@@ -14,6 +14,30 @@ module.exports = {
             serif: ['Noto Serif TC', '-apple-system', 'serif'],
         },
         extend: {
+            colors: {
+                'prim-100': '#F0EBF5',
+                'prim-300': '#AFAEE6',
+                'prim-500': '#9387FF',
+                'prim-700': '#6266CE',
+                'prim-800': '#352D66',
+                'prim-900': '#1F1C3B',
+                'black-200': '#CDD2E2',
+                'black-300': '#A8B4CA',
+                'black-500': '#20203A',
+                'black-700': '#1C1C38',
+                'black-800': '#1A1A30',
+                'black-900': '#121023',
+                'pink-500': '#C386AE',
+                'pink-700': '#E099E1',
+                'second-300': '#A9A6D6',
+                'second-700': '#9258BF',
+            },
+            stroke: {
+                'prim-100': '#F0EBF5',
+            },
+            boxShadow: {
+                'pink-500': '6px 6px 0px #C386AE',
+            },
```

Comment:
1.&2. :ok_hand:

3.There're no problems with the setups, and the codes look like this :

```js
colors: {
    transparent: 'transparent',
    current: 'currentColor',
    black: colors.black,
    white: colors.white,
    gray: colors.gray,
    indigo: colors.indigo,
    red: colors.rose,
    yellow: colors.amber,
    primary: {
        100: '#f0ebf5',
        300: '#afaee6',
        500: '#9387ff',
        700: '#6266ce',
        800: '#352d66',
        900: '#1f1c3b',
    },
    black: {
        200: '#cdd2e2',
        300: '#a8b4ca',
        500: '#20203a',
        700: '#1c1c38',
        800: '#1a1a30',
        900: '#121023',
    },
    pink: {
        500: '#c386ae',
        700: '#e099e1',
    },
    secondary: {
        300: '#a9a6d6',
        700: '#9258bf',
    },
},
stroke: theme => theme('colors'),
boxShadow: theme => ({
    'pink-500': `6px 6px 0 theme('colors.pink.500')`,
}),
```
is that right?

4.What does `define color palettes alone` mean? Does it need to create a new file and import to tailwind.config, just like `const colors = require('tailwindcss/colors')` ? if there're too many details you can also skip this question XD. Thanks for your reply.

Reply:
3. From your code block, if `stroke` is using the same color names, then I think we can omit the line of `theme => theme('colors')`. 
As for the `boxShadow`, beware of the variable in the string template.
Lastly, if we're keeping both tailwind's colors and our colors, simply put the `colors` under `extend`:
```js
module.exports = {
  mode: 'jit',
  ...,
  theme: {
    extend: {
      colors: {
        primary: {
          100: '#f0ebf5',
          300: '#afaee6',
          500: '#9387ff',
          700: '#6266ce',
          800: '#352d66',
          900: '#1f1c3b',
        },
        black: {
          200: '#cdd2e2',
          300: '#a8b4ca',
          500: '#20203a',
          700: '#1c1c38',
          800: '#1a1a30',
          900: '#121023',
        },
        pink: {
          500: '#c386ae',
          700: '#e099e1',
        },
        secondary: {
          300: '#a9a6d6',
          700: '#9258bf',
        },
      },
    },
  },
}
```

4. Scratch what I mentioned here. Sorry for the confusion. I didn't pay attention to the original ticket context and what's happening in Figma. Let's just follow what the design team has defined the color names in Figma. Otherwise, the naming conventions will be confusing across the teams.

Reply:
At first, I tried not  to add `stroke: theme => theme('colors')`, but it prints error :
```
(45:0) C:\IconHamburgerMenuIcon.vue The `stroke-primary-100` class does not exist. If `stroke-primary-100` is a custom class, make sure it is defined within a `@layer` directive.

  43 | <style scoped>
  44 | .core-iconHamburgerMenuIcon {
> 45 |     @apply stroke-primary-100;
     |                                                  ^
  46 | }
  47 | </style>
```
I try to use `@layer` , but no matter how I try, it prints the same error again, so I quit.
It can also write in css like :
```
.core-iconHamburgerMenuIcon {
    stroke: theme('colors.primary.100');
}
```
but I think setting in tailwind.config is better.
Is `@layer` the point?
thanks for your reply.

Reply:
Hmm so if you add the `stroke: theme => theme('colors')` will work without the errors? Keep it then. Sorry for the confusion, I mistook tailwind config's settings.

Reply:
Got it, thanks.